pkg/network/ebpf/c/tracer.h

```diff
@@ -56,6 +56,85 @@ typedef struct {
     tcp_stats_t tcp_stats;
 } tcp_conn_t;
 
+// From include/net/tcp.h
+// tcp_flag_byte(th) (((u_int8_t *)th)[13])
+#define TCP_FLAGS_OFFSET 13
+#define TCPHDR_FIN 0x01
+
+// skb_info_t embeds a conn_tuple_t extracted from the skb object as well as
+// some ancillary data such as the data offset (the byte offset pointing to
+// where the application payload begins) and the TCP flags if applicable.
+// This struct is populated by calling `read_conn_tuple_skb` from a program type
+// that manipulates a `__sk_buff` object.
+typedef struct {
+    conn_tuple_t tup;
+    __u32 data_off;
+    __u8 tcp_flags;
+} skb_info_t;
+
+// This determines the size of the payload fragment that is captured for each HTTP request
+#define HTTP_BUFFER_SIZE 25
+// This controls the number of HTTP transactions read from userspace at a time
+#define HTTP_BATCH_SIZE 15
+// The greater this number is the less likely are colisions/data-races between the flushes
+#define HTTP_BATCH_PAGES 10
+
+typedef enum {
+    HTTP_UNKNOWN           = 0,
+    HTTP_RESPONDING        = 1,
+    HTTP_REQUESTING_GET    = 1 << 1,
+    HTTP_REQUESTING_POST   = 1 << 2,
+    HTTP_REQUESTING_PUT    = 1 << 3,
+    HTTP_REQUESTING_DELETE = 1 << 4,
+    HTTP_REQUESTING_HEAD   = 1 << 5,
+} http_state_t;
```

Comment:
I mentioned this before, but I'll post it again:

This feels confusing, as it is mixing two concerns:

the state of the transaction: unknown->requesting->responding
The request method
It is also referenced in code as both state and packet_type. Perhaps they should be separate things, or at least different masked parts of the byte.

Method also doesn't need to be a bitmask, since it cannot be multiple methods at the same time.

Reply:
Done. Let me know if the code now is a bit more clear